pastebin_connector.py

```diff
@@ -1,194 +1,538 @@
-        print (json.dumps(json.loads(ret_val), indent=4))
-
-    exit(0)
+# File: pastebin_connector.py
+#
+# Copyright (c) 2019-2022 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+# Python 3 Compatibility imports
+import os
+import traceback
+import urllib.parse
+import uuid
+from urllib.parse import urlparse
+
+import phantom.app as phantom
+import phantom.rules as ph_rules
+import pytz
+import requests
+import simplejson as json
+from bs4 import BeautifulSoup
+from dateutil.parser import parse
+from django.utils.encoding import smart_str
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+from phantom.vault import Vault
+
+from pastebin_consts import *
+
+requests.packages.urllib3.disable_warnings()
+
+
+class RetVal(tuple):
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class PasteBinConnector(BaseConnector):
+
+    def __init__(self):
+        """Initialize global variables."""
+        # Call the BaseConnectors init first
+        super(PasteBinConnector, self).__init__()
+
+        self._state = {}
+        self._user_key = ""
+        self._api_key = ""
+        self._pastebin_username = ""
+        self._pastebin_password = ""
+
+    def _convert_to_client_tz(self, t_str):
+        # TODO: Next version will have a configurable TZ
+        t = parse(t_str, tzinfos={"CDT": -18000, "CST": -21600})
+        client_tz = pytz.UTC
+
+        # try:
+        #    client_tz = pytz.timezone("EST5EDT")
+        # except:
+        #    client_tz = pytz.UTC
+        return t.astimezone(pytz.timezone(client_tz.zone)).strftime("%Y-%m-%d %H:%M:%S %Z")
+
+    def _add_file_to_vault(self, action_result, file_name, container_id, file_data):
+        # action_result = ActionResult(dict(param))
+        success = True
+
+        if hasattr(Vault, 'get_vault_tmp_dir'):
+            temp_dir = Vault.get_vault_tmp_dir()
+        else:
+            temp_dir = "opt/phantom/vault/tmp"
+
+        temp_dir = temp_dir + '/{}'.format(uuid.uuid4())
+        os.makedirs(temp_dir)
+        fname = os.path.join(temp_dir, file_name)
+
+        try:
+            with open(fname, "w") as outfile:
+                outfile.write(smart_str(file_data))
+        except IOError:
+            success = False
+
+        if success:
+            # Vault.add_attachment(fname, container_id, file_name=None, metadata=dict())

```

Comment:
Same for this line, can we just remove the commented out code?

Reply:
Done.